manifests/init.pp

```diff
@@ -1,46 +1,35 @@
 # == Class keepalived
 #
 class keepalived (
-  $sysconf_dir        = $::keepalived::params::sysconf_dir,
-  $sysconf_options    = $::keepalived::params::sysconf_options,
-  $config_dir         = $::keepalived::params::config_dir,
-  $config_dir_mode    = $::keepalived::params::config_dir_mode,
-  $config_file_mode   = $::keepalived::params::config_file_mode,
-  $config_group       = $::keepalived::params::config_group,
-  $config_owner       = $::keepalived::params::config_owner,
-  $daemon_group       = $::keepalived::params::daemon_group,
-  $daemon_user        = $::keepalived::params::daemon_user,
-  $pkg_ensure         = $::keepalived::params::pkg_ensure,
-  $pkg_list           = $::keepalived::params::pkg_list,
-  $service_enable     = $::keepalived::params::service_enable,
-  $service_ensure     = $::keepalived::params::service_ensure,
-  $service_hasrestart = $::keepalived::params::service_hasrestart,
-  $service_hasstatus  = $::keepalived::params::service_hasstatus,
-  $service_manage     = $::keepalived::params::service_manage,
-  $service_name       = $::keepalived::params::service_name,
-  $service_restart    = $::keepalived::params::service_restart,
-  $vrrp_instance      = {},
-  $vrrp_script        = {},
-  $vrrp_sync_group    = {},
+  String[1] $sysconf_dir     = $keepalived::params::sysconf_dir,
+  String    $sysconf_options = $keepalived::params::sysconf_options,
+
+  Stdlib::Absolutepath $config_dir       = $keepalived::params::config_dir,
+  Stdlib::Filemode     $config_dir_mode  = $keepalived::params::config_dir_mode,
+  Stdlib::Filemode     $config_file_mode = $keepalived::params::config_file_mode,
+
+  String[1] $config_group = $keepalived::params::config_group,
```

Comment:
Why not `Variant[String[1], Integer]` ? This will permit to use UID also .

Reply:
It might be very rare anyone wants to do this and GID as a string does already work. Dunno if it just complicates the type with no real benefit.

Reply:
I am thinking about this because with root i always use 0.
But i am fine also with KIS.